### PR TITLE
M2P-241 Remove Amasty Store Credit from Bolt modal

### PR DESCRIPTION
**Pull request by dumega**
# Description
Add support to remove Amasty Store Credit from Bolt modal

Fixes: https://boltpay.atlassian.net/browse/M2P-241

#changelog Remove Amasty Store Credit from Bolt modal

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

File: ThirdPartyModules/Amasty/StoreCredit.php [hunk 1]
```diff
@@ -111,4 +112,53 @@ public function filterProcessLayout($result)
...
+        $websiteId,
+        $storeId
+    )
+    {
+        try {
+            if ($couponCode == self::AMASTY_STORECREDIT && $quote->getData(\Amasty\StoreCredit\Api\Data\SalesFieldInterface::AMSC_USE)) {
+                $amastyApplyStoreCreditToQuote->cancel($quote->getId());
+            }
+        } catch (\Exception $e) {
+            throw $e;
```
Comment (johnmccombs1) [inline ThirdPartyModules/Amasty/StoreCredit.php:161]: Can you log the exception to Bugsang?

**Review by johnmccombs1 — APPROVED** [review]
LGTM

**Review by vitaliyreznikov — APPROVED** [review]
@dumega I've tested, it works as expected. Could you please fix conflict?

File: ThirdPartyModules/Amasty/StoreCredit.php (hunk 1 above)
Comment (dumega) [reply ThirdPartyModules/Amasty/StoreCredit.php:161]: @johnmccombs1 It is there https://github.com/BoltApp/bolt-magento2/blob/master/Model/Api/UpdateCart.php#L298

**Review by danac-gs — APPROVED** [review]

**Review by vitaliyreznikov — APPROVED** [review]

**Pull request merged**